src/main/java/com/st/utopia/cancellationservice/dao/TicketDao.java

```diff
@@ -0,0 +1,49 @@
+package com.st.utopia.cancellationservice.dao;
+
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+//import org.springframework.data.jpa.repository.Modifying;
+//import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import com.st.utopia.cancellationservice.model.Ticket;
+import com.st.utopia.cancellationservice.model.TicketIdentity;
+
+@Repository
+public interface TicketDao extends JpaRepository<Ticket, TicketIdentity>{
+	/**
+	 * Get the ticket with a specific flight and row and seat in the plane .
+	 *
+	 */
+	default Ticket getTicket(Integer flight, Integer row, Character seat) {
+		final Optional<Ticket> record = findById(new TicketIdentity(flight, row, seat));
+		if (record.isPresent()) {

```

Comment:
Lines 21 through 25 can be condensed to `return record.orElse(null)`.

Reply:
Oh that is good to know and I agree